cli_flags.go

```diff
@@ -113,7 +113,7 @@ func parseArgs() (*arguments, error) {
 				Value:       "dd-otel-host-profiler",
 				Usage:       "Service name.",
 				Destination: &args.serviceName,
-				Sources:     cli.EnvVars("DD_HOST_PROFILING_SERVICE", "DD_SERVICE"),
+				Sources:     cli.EnvVars("DD_HOST_PROFILING_SERVICE"),
```

Comment:
Only do this when split by service is not enabled

Reply:
Right - it feels a bit weird to use `DD_SERVICE` for two distinct purposes:
* the profile service tag when split by service is not enabled, and
* the profiler's service name when split by service is enabled

Meaning if someone enables split by service without removing the `DD_SERVICE` environment variable, we'll set the service of the profiler itself to the node name.  I don't have a strong opinion, we could also leave it, and make sure we update the documentation to call that out.